account_id.go

```diff
@@ -201,11 +248,19 @@ func (id AccountID) Compare(given AccountID) int {
 		return -1
 	}
 
+	if id.AliasKey != nil && given.AliasKey != nil {
+		if id.AliasKey.String() > given.AliasKey.String() { //nolint
```

Comment:
I believe this is going to compare the pointers to the strings, not the actual values. I believe you need to use `strings.Compare(id.AliasKey.String(), given.AliasKey.String())` instead.

Reply:
no this should be comparing string to string, as thats what String() returns